woocommerce-gutenberg-products-block.php

```diff
@@ -113,3 +113,72 @@ function() {
 
 add_action( 'plugins_loaded', array( '\Automattic\WooCommerce\Blocks\Package', 'init' ) );
 
+/**
+ * Pre-filters script translations for the given file, script handle and text domain.
+ *
+ * @param string|false|null $translations JSON-encoded translation data. Default null.
+ * @param string|false      $file         Path to the translation file to load. False if there isn't one.
+ * @param string            $handle       Name of the script to register a translation domain to.
+ * @param string            $domain       The text domain.
+ * @return string JSON translations.
+ */
+function woocommerce_blocks_get_i18n_data_json( $translations, $file, $handle, $domain ) {
+	if ( 'woo-gutenberg-products-block' !== $domain ) {
+		return $translations;
+	}
+
+	global $wp_scripts;
+
+	if ( ! isset( $wp_scripts->registered[ $handle ], $wp_scripts->registered[ $handle ]->src ) ) {
+		return $translations;
+	}
+
+	$handle_filename = basename( $wp_scripts->registered[ $handle ]->src );
+	$locale          = determine_locale();
+	$lang_dir        = WP_LANG_DIR . '/plugins';
+
+	// Translations are always based on the unminified filename.
+	if ( substr( $handle_filename, -7 ) === '.min.js' ) {
+		$handle_filename = substr( $handle_filename, 0, -7 ) . '.js';
+	}
+
+	// WordPress 5.0 uses md5 hashes of file paths to associate translation
+	// JSON files with the file they should be included for. This is an md5
+	// of 'packages/woocommerce-blocks/build/FILENAME.js'.
+	$core_path_md5     = md5( 'packages/woocommerce-blocks/build/' . $handle_filename );
+	$core_json_file    = $lang_dir . '/woocommerce-' . $locale . '-' . $core_path_md5 . '.json';
+	$json_translations = is_file( $core_json_file ) && is_readable( $core_json_file ) ? file_get_contents( $core_json_file ) : false; // phpcs:ignore
```

Comment:
would prefer to see `// phpcs:ignore` on the line above this (according to docs it'll ignore the line it's on plus the line after).

Reply:
Just looking in woo core, we do them inline unless doing before/end rules for blocks of code.

Reply:
Sure, but it's [still valid to do `// phpcs:ignore` on the line before](https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file) and I would argue preferable to do so along with the specific rules being ignored so it's clear from reading the code.

Personally I think it'd be good to adopt it for Woo core as well for the following reasons:

- surfaces the comment more readily for the developer to see
- identifies clearly what rules are being ignored
- aligns with similar commenting standards use elsewhere (although I know WordPress standards doesn't enforce the location of the comment block).

I don't think it needs to block this pull though, it's just something I have opinions on :)

Reply:
I'd argue that these types are comments shouldn't be as visible to developers, it's not like they explain anything about the code itself, they are there to stop a machine complaining about something. Having it inline also makes it clear which line is affected, rather than wondering if the rule exclusion is for a block. But yes, doesn't block this :)

Reply:
Don't feel like you have to engage in replies if you don't want to, I just like to follow up on discussions like this when I feel I have something more to contribute to the convo.  I know it's a very opinionated subject (hence why I didn't think it blocked this pull).

> I'd argue that these types are comments shouldn't be as visible to developers, it's not like they explain anything about the code itself, they are there to stop a machine complaining about something. 

That's fair, in some cases they aren't necessarily warranted, however one problem I see with just doing a blanket ignore is that you essentially instruct the machine to ignore ALL linting for that line.  So that means any other rules that might pick up something will get ignored too.  Adding specificity keeps the linting for rules not being explicitly ignored (which applies not only to the current code, but also potential refactors of that line).  Admittedly there's often clear cases where it just doesn't matter, but habits are hard to establish, easy to follow, and hard to break 😄 .  It used to be a real pain to look up the rule and add it, but now most IDE's make it fairly easy to copy and paste the rule to be ignored.

>  Having it inline also makes it clear which line is affected, rather than wondering if the rule exclusion is for a block.

Yeah I could see that point of view and I think it works really well when the lines are shorter in length, but when the line gets long I think it becomes _less_ visible that a rule is being ignored. So my reasoning is rather than have exceptions to account for variations in visibility, just always put it directly before the line.  You can also address the issue with guessing what block it applies to by adding blank lines around the comment and the expression it applies to:

```php
$something = 10;

// phpcs: ignore
$my_non_linted_expression = 'howdy';

$something_else = 42;
```

Reply:
All good points. I see 3 things you're not happy about here:

1. About having the comment inline.
2. The readability of long lines of code
3. The specificity of the ignore rule.

(3) is good feedback and should have been done here, too late now, but still. We should definitely be specific so future linting rules are not inadvertently disabled by blanket ignores.

Personally I still prefer linting rules eol and I don't mind scrolling in my editor to see them. When I'm reading code I'm interested in what it's doing, not what rules were disabled at the time :)